src/main/java/com/nvs/store/service/ProductService.java

```diff
@@ -24,9 +23,9 @@ public Optional<Product> getProduct(Long id) {
         return productRepository.findById(id);
     }
 
-    public Product addProduct(Product product) {
-        productRepository.save(product);
```

Comment:
It would be nice to get rid of the null argument for Id. Perhaps you could remove the Restriction on the ID field from the Entity and the RequiredArgsContructor won't need this field anymore.